ompi/mca/coll/base/coll_base_bcast.c

```diff
@@ -798,6 +798,7 @@ int ompi_coll_base_bcast_intra_scatter_allgather(
     int recv_count = 0, send_count = 0;
     int scatter_count = (count + comm_size - 1) / comm_size; /* ceil(count / comm_size) */
     int curr_count = (rank == root) ? count : 0;
+    int rem_count = 0;
```

Comment:
I'm not sure I understand this. In C99 (which we require) variables can be declared anywhere.

Reply:
I don't think goto's are allowed to jump across declarations. gcc seems pretty lax on it, maybe because it's smart enough to know it won't change as currently written. But it could lead to undefined behavior in the future if someone overlooks that.

I see this on the top of page 139:
```
EXAMPLE 2    A goto statement  is  not  allowed  to  jump  past  any  declarations  of  objects  with  variably modified types.  A jump within the scope, however, is permitted.
http://www.dii.uchile.cl/~daespino/files/Iso_C_1999_definition.pdf
```

Reply:
If the jumped variables were used in the code after to goto label I would understand, but that's not the case, they remain used in the code block before the goto label. This i also exactly what the Example 2 you mentioned indicates, it is incorrect to jump the variable definition.

I hate to change a perfectly correct code just to please a particular compiler, but as humans are much more adaptable than software we will cope with the new code.

Reply:
I agree that in these cases there is no harm (the code is fine as is), but I think it's a good change in general. The compiler won't throw an error if someone were to try to use these uninitialized variables after the goto, and the developer will be left scratching their head.